### PR TITLE
Fix Host header issue, add options and support schemes

### PR DESCRIPTION
**Pull request by carlospolop**
- In new versions of "net/http" you cannot add a host header the way it's done in your current code (this: https://github.com/golang/go/issues/29865)
- I have added a couple of new params more
- If from stdin the protocol is given (http or https), it's used instead of trying both

In general the tool is fixed an improved without affecting how it worked previously. I hope you find the PR useful:)
And thank you for the tool!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
+
 # hakoriginfinder
 
 Tool for discovering the origin host behind a reverse proxy. Useful for bypassing WAFs and other reverse proxies.
 
+**This is a working and improved version!**
```
Comment (hakluke) [inline README.md]: Take this bit out

File: README.md [hunk 2]
```diff
@@ -55,5 +58,5 @@ NOMATCH https://1.1.1.2 19532
 Install golang, then run:
 
 ```
-go install github.com/hakluke/hakoriginfinder@latest
+go install github.com/carlospolop-forks/hakoriginfinder@latest
```
Comment (hakluke) [inline README.md]: It looks like you've changed the instructions to install it from your fork which is not ideal

File: go.mod [hunk 3]
```diff
@@ -1,3 +1,3 @@
-module github.com/hakluke/hakoriginfinder
+module github.com/carlospolop-forks/hakoriginfinder@latest
```
Comment (hakluke) [inline go.mod]: Same here

**Comment by hakluke** [comment]
Hey! Thanks for sorting these out! If you resolve those couple of things above ^ I'll merge it in!

**Comment by carlospolop** [comment]
Hey @hakluke! Happy new year!
Ofc, those changes shouldn't have been in the PR. I sent the PR before those changes, I didn't know PRs were automatically updated with changes after they are created.

I have removed that stuff:)
(And thank you for creating this tool)

**Review by hakluke — APPROVED** [review]

**Comment by hakluke** [comment]
Looks good - thank you!

**Comment by hakluke** [comment]
@carlospolop I reverted this because with the new PR the tool wasn't working... It gets a bunch of false matches. See below:

image

After reverting, the tool works as expected again:

image

**Comment by carlospolop** [comment]
Hi @hakluke,

I have no idea why you got those results. If you looked the code you might have noticed that I never touched the code for the Levenshtein distance calculator. The code you can see only adds functionality and fixes the Host header problem (your code didn't worked in a clean installation due to the error I shared with you at the begging).

This is my result using my version:
image

If you investigate more what happened there let me know your thoughts, because it's weird

**Comment by hakluke** [comment]
Could it be due to the http/https change? E.g if the http version returns a
30x redirect then it might match despite not being the right host?

**Comment by carlospolop** [comment]
If there is no http or https specified both are used (same behaviour as before).
The thing is that when I use my version of the tool I don't get the matches you got, so I don't know why isn't it working for you or how to debug that.

**Pull request merged**